frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+  const theme = useTheme()
+  data.forEach((entry: any) => {
+    entry = assign(entry, {
+      colorscale: hasLightBackgroundColor(theme)
+        ? sequentialColorscaleLightTheme
+        : sequentialColorscaleDarkTheme,
+    })
+  })
+  return data
+}
+
+/**
+ * This applies colors specifically for the table, candlestick, and waterfall plot
+ * because their dictionary structure is different from other more regular charts.
+ * @param data - spec.data
+ */
+export function applyUniqueGraphColorsData(data: any): void {
+  const theme = useTheme()
+  const { colors, genericFonts } = theme
+  data.forEach((entry: any) => {
+    // entry.type is always defined
+    if (entry.type === "table") {
+      entry.header = assign(entry.header, {
+        font: {
+          color: getGray70(theme),
+          family: genericFonts.bodyFont,
+        },
+        line: { color: colors.fadedText05, width: 1 },
+        fill: {
+          color: colors.bgMix,
+        },
+      })
+      entry.cells = assign(entry.cells, {
+        font: {
+          color: getGray90(theme),
+          family: genericFonts.bodyFont,
+        },
+        line: { color: colors.fadedText05, width: 1 },
+        fill: {
+          color: colors.bgColor,
+        },
+      })
+    } else if (entry.type === "candlestick") {
+      if (entry.decreasing === undefined) {
+        entry = assign(entry, {
+          decreasing: {
+            line: {
+              color: getDecreasingRed(theme),
+            },
+          },
+        })
+        entry = assign(entry, {
+          increasing: {
+            line: {
+              color: getIncreasingGreen(theme),
+            },
+          },
+        })
+      }
+    } else if (entry.type === "waterfall") {
+      entry = assign(entry, {
+        connector: {
+          line: {
+            color: getGray30(theme),
+            width: 2,
+          },
+        },
+      })
+      if (entry.decreasing === undefined) {
+        entry = assign(entry, {
+          decreasing: {
+            marker: {
+              color: getDecreasingRed(theme),
+            },
+          },
+          increasing: {
+            marker: {
+              color: getIncreasingGreen(theme),
+            },
+          },
+          totals: {
+            marker: {
+              color: hasLightBackgroundColor(theme) ? "#0068C9" : "#83C9FF",
+            },
+          },
+        })
+      }
+    }
+  })
+}
+
+/**
+ * This applies general layout changes to things such as x axis,
+ * y axis, legends, titles, grid changes, background, etc.
+ * @param layout - spec.layout.template.layout
+ * @param theme - Theme from useTheme()
+ */
+export function applyStreamlitThemeTemplateLayout(
+  layout: any,
+  theme: Theme
+): void {
+  const { genericFonts, colors, fontSizes } = theme
+
+  const streamlitTheme = {
+    // hide all text that is less than 8 px
+    uniformtext: {
+      minsize: 8,
+      mode: "hide",
+    },
+    background: colors.bgColor,
```

Comment:
Are you sure this is a valid plotly property? I cannot find it in the docs [here](https://plotly.com/python/reference/layout/#layout-template).